lib/dpl/provider.rb

```diff
@@ -57,6 +57,7 @@ def self.new(context, options)
         name = super.option(:provider).to_s.downcase.gsub(/[^a-z0-9]/, '')
         raise Error, 'could not find provider %p' % options[:provider] unless name = constants.detect { |c| c.to_s.downcase == name }
         provider = const_get(name).new(context, options)
+        puts "Provider #{name} with options #{options}"
```

Comment:
Not sure if we can add this unconditionally. While we do have filter to suppress secrets, some information may be best left unwritten.